### PR TITLE
Gh-3317: Addressing the seven issues raised in SonarCloud.

### PR DESCRIPTION
**Pull request by rj77259**

### CODE REVIEW

File: store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/operation/hdfs/mapper/SampleDataForSplitPointsMapper.java [hunk 1]
```diff
@@ -63,7 +65,7 @@ protected void setup(final Context context) {
 
     @Override
     protected void map(final Element element, final Context context) throws IOException, InterruptedException {
-        if (Math.random() < proportionToSample) {
+        if (secureRandom.nextDouble() < proportionToSample) {
```
Comment (tb06904) [inline store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/operation/hdfs/mapper/SampleDataForSplitPointsMapper.java:68]: Again this is just used to create samples out of the data there's no need for secure random.

File: core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/AbstractSampleElementsForSplitPointsHandler.java [hunk 2]
```diff
@@ -27,16 +27,17 @@
...
 import java.util.List;
-import java.util.Random;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 public abstract class AbstractSampleElementsForSplitPointsHandler<T, S extends Store> implements OutputOperationHandler<SampleElementsForSplitPoints<T>, List<T>> {
     public static final int MAX_SAMPLED_ELEMENTS_DEFAULT = 10000000;
 
     private int maxSampledElements = MAX_SAMPLED_ELEMENTS_DEFAULT;
+    private static SecureRandom secureRandom = new SecureRandom();
```
[the remark this answers is not in the document]
Comment (rj77259) [reply core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/AbstractSampleElementsForSplitPointsHandler.java:40]: PMD suggestion at the time of development will double check this was a false positive. Recommendation to put it here

File: store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/operation/hdfs/mapper/SampleDataForSplitPointsMapper.java (hunk 1 above)
Comment (rj77259) [reply store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/operation/hdfs/mapper/SampleDataForSplitPointsMapper.java:68]: Can revert back makes sense

File: library/spark/spark-accumulo-library/src/main/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/handler/javardd/SplitStoreFromJavaRDDOfElementsHandler.java [hunk 3]
```diff
@@ -32,16 +32,18 @@
...
 import uk.gov.gchq.gaffer.store.schema.Schema;
 
+import java.security.SecureRandom;
 import java.util.Iterator;
 import java.util.List;
-import java.util.Random;
 
 import static java.util.Arrays.asList;
 
+
```
[the remark this answers is not in the document]
Comment (rj77259) [reply library/spark/spark-accumulo-library/src/main/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/handler/javardd/SplitStoreFromJavaRDDOfElementsHandler.java:41]: Will remove

**Comment by rj77259** [comment]
Closing and flagging this in Sonarcloud

**Pull request closed without merging**